### ISSUE TITLE
[Refactor] Introduce internal `Auto*Operation` base classes across `keras.ops` to reduce boilerplate

### ISSUE THREAD

**Issue by james77777778** [keras-team/keras#23625]
I already have a draft commit for this functionality: https://github.com/james77777778/keras/commit/2daef84780639d7bbf40eda159cbc7f254ace4a4
Feel free to assign this to me.

### Problem Description

Across `keras/src/ops` (`numpy.py`, `nn.py`, `math.py`, `core.py`), dozens of operations are thin wrappers around backend functions. Currently, each operation re-implements the exact same boilerplate.

This duplication increases maintenance overhead and makes it difficult to maintain uniform symbolic propagation rules (such as consistent `sparse` and `ragged` handling) across all operations.

### Proposed Solution

Introduce three internal base classes in `keras/src/ops/operation.py` that encapsulate standard execution and symbolic shape/dtype inference:

1. **`AutoElementwiseOperation`**
   - For unary elementwise ops (activations, math functions, unary predicates).
   - Automatically dispatches `call(*args, **kwargs)` to `self.backend_fn`.
   - Infers `compute_output_spec()` by matching input shape and preserving `sparse` and `ragged` attributes. Supports an optional `output_dtype` override (e.g. `"bool"` for predicates like `isnan`).

2. **`AutoBinaryBroadcastOperation`**
   - For binary broadcasting operations (arithmetic, comparisons, bitwise ops).
   - Dispatches `call(x1, x2)` to `self.backend_fn`.
   - Computes output shape via `broadcast_shapes(x1.shape, x2.shape)`, infers result dtype via `dtypes.result_type`, and preserves `sparse` and `ragged` tensor semantics.

3. **`AutoReductionOperation`**
   - For axis reductions (`amax`, `amin`, `all`, `any`, `argmax`, `argmin`, etc.).
   - Standardizes `__init__(self, axis=None, keepdims=False, *, name=None)` and delegates `call(x)`.
   - Computes output shape via `reduce_shape()` and correctly handles `output_dtype` (e.g., `"bool"` for `all`/`any`, `"int32"` for `argmax`/`argmin`).

#### Example Usage

Ops become clean and declarative:

```python
class Relu(AutoElementwiseOperation):
    backend_fn = backend.nn.relu

class Equal(AutoBinaryBroadcastOperation):
    backend_fn = backend.numpy.equal
    output_dtype = "bool"

class Amax(AutoReductionOperation):
    backend_fn = backend.numpy.amax
```

### Scope & Impact

- Refactors **80 operations** across `core.py`, `math.py`, `nn.py`, and `numpy.py` (~24% of all op classes in `keras.ops`).
- Eliminates **over 700 lines** of duplicate boilerplate.
- Standardizes symbolic handling for `ragged` and `sparse` tensors.

**Comment by maitry63** [comment]
Hi @james77777778,

Thanks for reporting this and for your contribution. I went through the issue and your claims are correct.

Added a reproduction covering the key behavior described in the issue, including `elementwise operations`, `binary broadcasting`, `dtype inference`, `reductions`, `keepdims`, and `concrete backend execution`.

Refer the  reproduction : [Colab](https://colab.sandbox.google.com/gist/maitry63/12944490b93e946951edb16778d5d367/repro_-refactor-introduce-internal-auto-operation-base-classes-across-keras-ops-to-reduce-boilerplate-23625.ipynb)

I'm assigning this issue to you. Thanks again for the contribution!
